### ISSUE TITLE
[no-magic-numbers] Computed constant values

### ISSUE THREAD

**Issue by mustafa-cosar** [typescript-eslint/typescript-eslint#1171]
**Repro**

```JSON
{
  "rules": {
   "'no-magic-numbers": "off",
    "@typescript-eslint/no-magic-numbers": [
            "error",
            {
                "ignore": [-1, 0, 1, 2],
                "ignoreEnums": true,
                "ignoreNumericLiteralTypes": true,
                "ignoreReadonlyClassProperties": true
            }
        ]
  }
}
```

```TS
export class Utilities
{
    public static readonly MAX_IMAGE_SIZE_IN_BYTES = 10 * 1024 * 1024;
}
```

**Expected Result**
No errors since the result is ultimately a constant number.

**Actual Result**
`No magic number` error for 10, 1024, 1024

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.5.0` |
| `@typescript-eslint/parser`        | `2.5.0` |
| `TypeScript`                       | `3.6.4` |
| `ESLint`                           | `6.5.1` |
| `node`                             | `v10.16.3` |
| `npm`                              | `6.12.0` |

**Comment by bradzacher** [comment]
This occurs in the logic in the base rule
https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLW1hZ2ljLW51bWJlcnM6IFtcImVycm9yXCIsIHtcbiAgICAgICAgICAgICAgICBcImlnbm9yZVwiOiBbLTEsIDAsIDEsIDJdLFxuICAgICAgICAgICAgfV0gKi9cblxuY29uc3QgIE1BWF9JTUFHRV9TSVpFX0lOX0JZVEVTID0gMTAgKiAxMDI0ICogMTAyNDsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEwLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7ImpzeCI6dHJ1ZX19LCJydWxlcyI6e30sImVudiI6eyJjb21tb25qcyI6ZmFsc2UsIm5vZGUiOmZhbHNlfX19

If you would like this fixed, please raise an issue in the eslint repo.

**Comment by mustafa-cosar** [comment]
Thanks for your help. I will open an issue on eslint repo.
